WS17_18/05_lifecycle_segues/your assignment/hahn/Yatodoa/Yatodoa/Controller/DetailTodoTableViewController.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  DetailTodoTableViewController.swift
+//  Yatodoa
+//
+//  Created by Alex on 28.11.17.
+//  Copyright © 2017 Alexander Dobrynin. All rights reserved.
+//
+
+import UIKit
+
+protocol DetailTodoTableViewControllerDelegate {
+    func dvc(_ vc: DetailTodoTableViewController, updatedTodo todo: ToDo)
+}
+
+class DetailTodoTableViewController: UITableViewController {
+    
+    private struct Storyboard {
+        static let SegueIdentifier = "NotesSegueIdentifier"
+    }
+    
+    private struct Constants {
+        static let Formatter: DateFormatter = {
+            let f = DateFormatter()
+            f.dateStyle = .short
+            f.timeStyle = .short
+            return f
+        }()
+        
+        static let NotesDefaultText = "Notiz..."
+    }
+    
+    // MARK: 1. Preparation
+    var todo: ToDo?
+    
+    var delegate: DetailTodoTableViewControllerDelegate?
+    
+    // MARK: 2. Outlets
+    @IBOutlet weak var tagLabel: UILabel!
+    @IBOutlet weak var notesLabel: UILabel!
+    
+    // MARK: 3. VC Lifecycle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        title = todo!.task
+        tagLabel.text = todo!.tag
+        notesLabel.text = todo!.notes ?? Constants.NotesDefaultText
+        
+        navigationController?.delegate = self
+    }
+    
+    override func tableView(_ tableView: UITableView, titleForFooterInSection section: Int) -> String? {
+        return section == 1 && todo!.completed ? "Erledigt am \(Constants.Formatter.string(from: todo!.finished!))" : nil
+    }
+    
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        guard let identifier = segue.identifier else { return }
+        
+        switch identifier {
+        case "TagsSegue":
```

Comment:
Segue Identifier sollten immer in Konstanten ausgelagert werden, um nur einen Punkte zu haben, diese zu ändern. Außerdem ist es übersichtlicher, siehe Zeile 18.